lib/mindbody-api/client.rb

```
@@ -23,14 +31,22 @@ def call(operation_name, locals = {}, &block)
       def auth_params
         {'SourceCredentials'=>{'SourceName'=>MindBody.configuration.source_name,
                                'Password'=>MindBody.configuration.source_key,
-                               'SiteIDs'=>{'int'=>MindBody.configuration.site_ids}}}
+                               'SiteIDs'=>{'int'=>MindBody.configuration.site_ids}},
+         'UserCredentials'=>{'Username'=>'_' + MindBody.configuration.source_name,
+                             'Password'=>MindBody.configuration.source_key,
+                             'SiteIDs'=>{'int'=>MindBody.configuration.site_ids}}}
       end
 
       def fixup_locals(locals)
         # TODO this needs fixed to support various list types
         locals.each_pair do |key, value|
           if value.is_a? Array
-            locals[key] = {'int' => value}
+            case key
+            when 'ClientIDs'
+              locals[key] = {'string' => value}
+            else
+              raise "Does not know what to do with #{key}"
```

Why raise instead of silently ignoring here?